frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js

```diff
@@ -68,67 +66,69 @@ const propTypes = {
   history: PropTypes.shape({ push: PropTypes.func }).isRequired,
 };
 
-const joinOrRemove = (param, key) => (isEmpty(param) ? {} : { [key]: param.join(",") });
-const formatParams = ({ region = [], compliance_code = [], ...remainingParams }) => ({
-  ...joinOrRemove(region, "region"),
-  ...joinOrRemove(compliance_code, "compliance_code"),
-  ...remainingParams,
-});
-
+export const joinOrRemove = (param, key) => {
+  if (isEmpty(param)) {
+    return {};
+  }
+  return typeof param === "string" ? { [key]: param } : { [key]: param.join(",") };
+};
+export const removeEmptyStings = (param, key) => (isEmpty(param) ? {} : { [key]: param });
+export const formatParams = ({
+  region = [],
+  compliance_code = [],
+  issue_date_after,
+  issue_date_before,
+  expiry_date_before,
+  expiry_date_after,
+  search,
+  major,
+  ...remainingParams
+}) => {
+  return {
+    ...joinOrRemove(region, "region"),
+    ...joinOrRemove(compliance_code, "compliance_code"),
+    ...removeEmptyStings(issue_date_after, "issue_date_after"),
+    ...removeEmptyStings(issue_date_before, "issue_date_before"),
+    ...removeEmptyStings(expiry_date_before, "expiry_date_before"),
+    ...removeEmptyStings(expiry_date_after, "expiry_date_after"),
+    ...removeEmptyStings(search, "search"),
+    ...removeEmptyStings(major, "major"),
+    ...remainingParams,
+  };
+};
 export class VarianceHomePage extends Component {
+  params = queryString.parse(this.props.location.search);
+
   constructor(props) {
     super(props);
+    const formatedParams = {
+      compliance_code: formatParamStringToArray(this.params.compliance_code),
+      region: formatParamStringToArray(this.params.region),
+      major: this.params.major,
+      search: this.params.search,
+      issue_date_after: this.params.issue_date_after,
+      issue_date_before: this.params.issue_date_before,
+      expiry_date_before: this.params.expiry_date_before,
+      expiry_date_after: this.params.expiry_date_after,
+    };
```

Comment:
Is there a reason for pulling this out into a variable?

Reply:
I think this was legacy of trying to fix the clear bug.  I have put it back